docs/operations.md

```diff
@@ -18,10 +18,54 @@ then deleting the old chunks. The scavenged events are also removed from the ind
 
 Running a scavenge causes the active chunk to be completed so that it can be scavenged.
 
-::: warning 
+::: warning
 Scavenging is destructive. Once a scavenge has run, you cannot recover any deleted events except from a backup.
 :::
 
+### Mechanism of the scavenging
+
+Running scavenge for the first time will take longer than usual because it needs to create a "scavenge point", then execute all phases, create folders and SQL database files, and finally mark the current chunk as completed before removing it.
+
+Whenever a scavenging process starts on a node, it checks if there are any existing unused scavenge point on that. If there are, it utilizes one of them; otherwise, it creates a new scavenge point and replicates it to another node.
+
+_Most of the time, the scavenging process does not run on the leader node to reduce the workload on the leader. It starts on one of those nodes and send the information back to the leader._
```

Comment:
Remove "_Most of the time, the scavenging process does not run on the leader node to reduce the workload on the leader. It starts on one of those nodes and send the information back to the leader._" and add a best practice section to advise running scavenging on Follower nodes and details there.